### DISCUSSION TITLE
I2C problems

### DISCUSSION THREAD

**Discussion by MikeMcPherson** [alanbjohnston/CubeSatSim#148] [Q&A] [answered]
Hi folks.  I've built the Main Board (V1.1) for the first of two CubeSat Sims.  The Pi Zero W boots, the pretty lights blink, and it transmits very nice telemetry that FoxTelem decodes with no trouble.

My problem is that I'm getting an error from the "CubeSatSim/telem" command saying there is a problem with I2C bus 1.  As far as I can tell, everything compares correctly with the instructions and the pictures.  Any thoughts on where I might look for the problem?

Thanks.  Mike KQ9P

**Comment by alanbjohnston** [comment]
Hi Mike,

If you get that error, it means that one of the INA219 boards on the I2C bus 1 has a problem.  

Are you using the purple INA219 boards or the blue ones?

Are you using plug-in sockets, or are they soldered into the board.

Check all your soldering to make sure there isn't a solder bridge between two pins, or that a board isn't mounted upside.   Also, check the A0 and A1 jumper settings on each of the INA219 boards 

You can try this command, too:

i2cdetect -y 1

It should show you four devices within a few seconds.

If your INA219 boards are plugged into sockets, you should unplug these four ones:  +X, +Y, Batt, 5V bus.  If the problem goes away, it is one of the boards.  You can try plugging them in one at a time to see when the problem happens.

73, Alan KU2Y

**Reply by MikeMcPherson** [reply]
Thanks for the suggestions, Alan.  Actually, none of the INA219 boards are in, yet.

I'm using the purple INA219s, and I plan to solder them in.

Thanks.  Mike KQ9P

**Reply by alanbjohnston** [reply]
What step in the assembly process are you at, then?  If there are no INA219s installed, then this error suggests a software problem. 

Did you start from a Raspberry Pi image or did you manually install the software?  If manual, what image did you start with? 

You could try manually enabling I2C by running `sudo raspi-config` Then selecting Interface Options then I2C then Enable. 

If you plan to solder in your purple INA219 boards, make sure you test them first as described in the instructions as there are lots of bad ones being sold these days. That’s why I have switched to the larger blue INA219s even though their fit is a bit awkward.

**Comment by MikeMcPherson** [comment]
I was just at the testing the INA219s stage (build step 4.2).  I started with the pre-built ISO from http://cubesatsim.org/download/cubesatsim-v1.2.iso.gz, then "apt update/upgrade" and reboot for Linux, then "CubeSatSim/update" and reboot.

I just thought I would ask in case this was a well-known issue that others have already solved.  I'll roll up my sleeves and start debugging...

**Reply by MikeMcPherson** [reply]
I discovered that pins 5 and 37 on the Pi header were shorted to ground; neither of them should be, and pin 5 should be, of course, pulled up to 3.3V.  I checked for solder bridges and found none.  My best guess was that there might be bridges underneath the stacking header, so I pulled out the second Main Board.  

Before populating, pins 5 and 37 were as they should be: not grounded.  I installed the pushbutton, microUSB breakout, remove before flight jack, and the three LEDs and resistors.  Now pins 5 and 37 are grounded again!

Looking at the schematic, I really don't understand how installation of those components could ground those two GPIO pins.  I'm going to look at the Gerbers to see if anything makes sense there.

Thanks.  Mike KQ9P

**Reply by alanbjohnston** [reply]
Pins 5 and 37 connect to the push button S1, so I would look closely at the soldering on it or maybe the switch is stuck.

**Comment by MikeMcPherson** [comment, marked as answer]
OK, figured it out.  When I ordered the parts the pushbutton switch in the BOM was (and still is) unobtainable.  I substituted another switch, but the one I chose has a particularly inscrutable datasheet and I read it wrong.  (Nowhere on the datasheet does it show the pin numbering for the DPDT version.)  The pins are backward.  I'm going to pull the switch and I'll bet things will work fine.

Thanks!  Mike KQ9P

**Reply by alanbjohnston** [reply]
Ah ha! I was notified by another builder that the push button switch S1 was out of stock so I added a link to Mouser in the BOM, so for now you can order it there.

**Comment by MikeMcPherson** [comment]
Thanks, grabbed some before they run out.  It's like the world's biggest scavenger hunt trying to find parts these days!

Mike KQ9P

**Comment by MikeMcPherson** [comment]
Now I'm wrestling with the dead INA219 problem.  So far, I've tested four and they've all been useless.  Argh!

**Reply by alanbjohnston** [reply]
When there were shortages of them, I think the manufacturer stopped QC testing and just shipped all of them. I’ve had much better luck since I switched to the blue INA219 boards - the latest BOM has them from Amazon at a reasonable price and fast shipping. Good luck!

**Reply by MikeMcPherson** [reply]
The blue breakouts arrived today.  So far I've tested 9 of them; all nine work perfectly.  This is much better!
